### ISSUE TITLE
Add priority settings to Runs

### ISSUE THREAD

**Issue by rhliang** [cfe-lab/Kive#508]
Jeff (@greenbeard1) has some runs that he started yesterday that aren't being done right now because he's caught behind a wave of MiCall tasks.  It would be good if we could set the priorities of the jobs we add to the queue, akin to how one calls a background task with the `nice` command.  We could use this for MiCall as well: new runs get normal priority, while verification re-runs get low priority.

Once this is done, we can raise the target number of reruns added for cfe-lab/micall#282.

**Comment by ArtPoon** [comment]
Look into using torque to schedule tasks, which will provide prioritization features as well as other resource management.

**Comment by rhliang** [comment]
This is a big undertaking so if we find ourselves able to push v0.8 sooner than later, we can push this to v0.9.

**Comment by rhliang** [comment]
A challenger approaches!  Compare Torque to Slurm.

**Comment by rhliang** [comment]
I've been reading up on Slurm and Torque, and there are several "cheatsheet"-type documents out there that indicate that they're essentially command-for-command functionally equivalent, to the point that a thin translation layer might be enough to move from one to the other.  Perhaps the fact that Slurm is GPL and Moab/Torque are proprietary is enough to be a deciding factor?

Maybe a better determining factor would be popularity of each platform.  Googling "slurm resource manager" gives just shy of 12000 hits, whereas "torque resource manager" yields more than 10x that.  It does seem like Slurm is trending upward, though -- but that could be that I'm too much in a Slurm echo chamber to make a proper judgment.

**Comment by rhliang** [comment]
We're going to use Slurm over Torque.  Next step is to do something about it.

**Comment by rhliang** [comment]
See #634 for discussion on priority policy.

**Comment by rhliang** [comment]
Backend support for priority is implemented (not tested yet!) in the Slurm branch.

**Comment by rhliang** [comment]
From a UI perspective it would be better to call this "Work Queue" rather than "Priority" because the queue can be tied to things other than priorities, e.g. resources.

In the future, we may go from an integer "priority" field to a string "queue" field.  This will entail creating a "fallback" Slurm queue field so that any Runs assigned to queues that disappear can be mapped to the fallback queue.

Check whether Slurm queues can be made exclusive to select accounts, so that `kivefleet` could perhaps be the exclusive user of some predefined partitions that are reserved for Kive.

**Comment by rhliang** [comment]
Closing because the backend functionality is complete; issue #672 will be moved to the next milestone.
